druid/src/core.rs

```diff
@@ -423,15 +423,25 @@ impl<T: Data, W: Widget<T>> WidgetPod<T, W> {
             LifeCycle::HotChanged(_) => false,
             LifeCycle::RouteFocusChanged { old, new } => {
                 self.state.request_focus = None;
-                let this_changed = old.map(|_| false).or_else(|| new.map(|_| true));
+
+                let this_changed = if *old == Some(self.state.id) {
```

Comment:
I think this respelling is much better. It took me several minutes tease apart the old version the first time.